### DISCUSSION TITLE
How to make tty work

### DISCUSSION THREAD

**Discussion by jonnitto** [deployphp/deployer#2976] [Help needed]
I'm working on updating my package from v6 to v7. I've some tasks that use `tty`, and I know it was completely removed from v7. These tasks are mainly helpers to edit configuration files directly with deployer, so that the integrator doesn't has to worry about the correct path. For example, how can I make `run('EDITOR={{editor}} crontab -e', ['timeout' => null, 'tty' => true]);` working with v7?

**Comment by antonmedv** [comment]
It's no longer possible with v7. tty was removed. As a workaround use Symfony/Process directly: new Process(ssh crontab, tty=>true)

**Comment by jonnitto** [comment]
I've written a function like this:

```php
function tty(string $command): void
{
    $command = parse($command);
    $host = currentHost();
    $connectionString = $host->getConnectionString();
    $ssh = ['ssh', '-A', '-tt', $connectionString, $command];
    $process = new Process($ssh);
    $process->setTimeout(null)->setIdleTimeout(null)->setTty(true);
    $process->run();
}
```
This works so far, but I can't make any input, and there's is also the indicator (see screenshot). How I can get rid of this?

![CleanShot 2022-02-28 at 12 35 52@2x](https://user-images.githubusercontent.com/4510166/155977170-360a60c4-567c-4779-82bd-f74c4265d630.png)

**Reply by antonmedv** [reply]
I see. There’s more difficulties. Now every task in Deployer runs in parallel mode means there is another layer of indirections. 

Without adding this feature to Master class it will not work. 

I’ll think on adding something like ttyTask().

**Comment by helloiamlukas** [comment]
+1

This is the only thing holding me back from updating to v7 currently.
